File: frontend/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpErrorResponse } from '@angular/common/http';
import { Observable, throwError } from 'rxjs';
import { User } from '../user';
import { UpdatePasswordDto } from '../update.pasword.dto';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private readonly API_URL = 'http://localhost:9000/api/users';

  constructor(private httpClient: HttpClient) {}

  getUser(): Observable<User> {
    return this.httpClient.get<User>(`${this.API_URL}`);
  }

  updateUser(user: User): Observable<any> {
    return this.httpClient.post(this.API_URL, user);
  }

  updatePassword(userId: string, updatePasswordDto: UpdatePasswordDto): Observable<any> {
    return this.httpClient.put(`${this.API_URL}/updatePassword/${userId}`, updatePasswordDto);
  }

}

File: user-service/src/main/java/userservices/service/UserService.java
package userservices.service;

import org.service.userservices.dto.PasswordUpdateRequest;
import org.service.userservices.dto.User;
import org.service.userservices.dto.UserUpdateRequest;

public interface UserService {

    User getUserInfo(String token);

    void updateUserInfo(UserUpdateRequest userUpdateRequest, String token);

    void updatePassword(PasswordUpdateRequest passwordUpdateRequest, String token, String userId);

}
